curated_transformers/layers/attention.py

```diff
@@ -148,12 +145,34 @@ def filter_batch_items(self, mask: Tensor) -> "AttentionMask":
         return AttentionMask(self.bool_mask[mask])
 
     def dim(self) -> int:
+        """
+        Return the number of dimensions in the mask.
+        """
         return self.bool_mask.dim()
 
     def merge_mask(self, other: "AttentionMask") -> "AttentionMask":
+        """
+        Merge another attention mask with this one.
+
+        :param other:
+            Attention mask to merge.
+        :returns:
+            Merged mask.
+        """
         return AttentionMask(self.bool_mask.logical_and(other.bool_mask))
 
-    def logit_mask(self, dtype: torch.dtype):
+    def logit_mask(self, dtype: torch.dtype) -> Tensor:
+        """
+        Generate the logit mask for the given ``dtype``.
+
+        Elements of the mask that are ``False`` are set to the
+        minimum value of the ``dtype`` and the rest to zero.
```

Comment:
We should probably add something saying that when used in softmax, that _adding_ this mask to the logits results in (near-)zero probabilities for the elements that are ``False``.